runhouse/main.py

```diff
@@ -679,6 +303,207 @@ def cluster_list(
         print_sky_clusters_msg(len(sky_clusters))
 
 
+@cluster_app.command("keep-warm")
+def cluster_keep_warm(
+    cluster_name: str = typer.Argument(
+        None,
+        help="Name of cluster to keep warm. If not specified will check the local cluster.",
+    ),
+    mins: Optional[int] = typer.Option(
+        -1,
+        help="Amount of time (in min) to keep the cluster warm after inactivity. If set to -1, keeps cluster warm indefinitely.",
+    ),
+):
+    """Keep the cluster warm for given number of minutes after inactivity.
+
+    Example:
+
+        ``$ runhouse cluster keep-warm /sashab/rh-basic-cpu``
+
+    """
+
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        if not current_cluster.is_up():
+            print_bring_cluster_up_msg(cluster_name=cluster_name)
+            return
+        current_cluster.keep_warm(mins=mins)
+    except ValueError:
+        console.print(f"{cluster_name} is not saved in Den.")
+        sky_live_clusters = get_all_sky_clusters()
+        cluster_name = (
+            cluster_name.split("/")[-1] if "/" in cluster_name else cluster_name
+        )
+        if cluster_name in sky_live_clusters:
+            console.print(
+                f"You can keep warm the cluster by running [italic bold] sky autostop {cluster_name} -i {mins}"
+            )
+    except Exception as e:
+        console.print(f"Failed to keep the cluster warm: {e}")
+
+
+@cluster_app.command("up")
+def cluster_up(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to bring up.",
+    )
+):
+    """Bring up the cluster if it is not up. No-op if cluster is already up.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+
+        ``$ runhouse cluster up /sashab/rh-basic-cpu``
+
+    """
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        current_cluster.up_if_not()
+    except ValueError:
+        console.print("Cluster is not saved in den.")
+    except Exception as e:
+        console.print(f"Failed to bring up the cluster: {e}")
+
+
+@cluster_app.command("down")
+def cluster_down(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to terminate.",
+    ),
+    remove_configs: bool = typer.Option(
+        False,
+        "-rm",
+        "--remove-configs",
+        help="Whether to delete cluster config from Den (default: ``False``).",
+    ),
+    remove_all: bool = typer.Option(
+        False,
+        "-a",
+        "--all",
+        help="Whether to terminate all running clusters saved in Den (default: ``False``).",
+    ),
+    confirm_deletion: bool = typer.Option(
```

Comment:
i think this is more clear about the implications of providing -y, wdyt?
```suggestion
    force_deletion: bool = typer.Option(
```